### PR TITLE
Hide archived assay designs

### PR DESCRIPTION
**Pull request by RosalineP**
#### Rationale
The purpose of this story is to provide a way for admins to archive an assay design. To this ends, we added a new 'Status' checkbox input to the Assay Designer Properties panel. When status is 'off', the assay design is considered to be archived, and does not appear in certain areas of the UI, such as the mega-menu or the new 'Active' tab of the Assays List page.

#### Related Pull Requests
* https://github.com/LabKey/commonAssays/pull/379
* https://github.com/LabKey/sampleManagement/pull/732
* https://github.com/LabKey/biologics/pull/1036
* https://github.com/LabKey/labkey-ui-components/pull/659

#### Changes
* (For Karl to populate)

### CODE REVIEW

File: assay/src/org/labkey/assay/AssayDomainServiceImpl.java [hunk 1]
```diff
@@ -204,6 +204,7 @@ private GWTProtocol getAssayTemplate(AssayProvider provider, Pair<ExpProtocol, L
         result.setName(protocol.getName());
         result.setProviderName(provider.getName());
         result.setDescription(protocol.getDescription());
+        result.setStatus(protocol.getStatus() != null ? protocol.getStatus().name() : ExpProtocol.Status.Active.name());
```
Comment (RosalineP) [inline assay/src/org/labkey/assay/AssayDomainServiceImpl.java:207]: A perhaps slightly hacky addition of mine to get create-case working

File: api/src/org/labkey/api/assay/AssayProvider.java [hunk 2]
```diff
@@ -138,7 +138,7 @@ default String getLabel()
     @NotNull
     AssayTableMetadata getTableMetadata(@NotNull ExpProtocol protocol);
 
-    ExpProtocol createAssayDefinition(User user, Container container, String name, String description) throws ExperimentException;
+    ExpProtocol createAssayDefinition(User user, Container container, String name, String description, @Nullable ExpProtocol.Status status) throws ExperimentException;
```
Comment (labkey-nicka) [inline api/src/org/labkey/api/assay/AssayProvider.java:141]: @labkey-klum I guess I'm a little surprised this needed to bubble up all the way to the `AssayProvider` interface. Isn't there other properties that can be set on the `ExpProtocol` that aren't configured via this top-level creation method?

**Review by labkey-nicka — APPROVED** [review]

File: assay/src/org/labkey/assay/AssayDomainServiceImpl.java (hunk 1 above)
Comment (labkey-nicka) [reply assay/src/org/labkey/assay/AssayDomainServiceImpl.java:207]: This is essentially how we configure the default value to be "Active". Seems fine.

File: api/src/org/labkey/api/assay/AssayProvider.java (hunk 2 above)
Comment (labkey-klum) [reply api/src/org/labkey/api/assay/AssayProvider.java:141]: Not that I found, we could set it directly on the returned `ExpProtocol` object but that would have required a second save to the database.

**Pull request merged**